demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1035 @@
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    lBytesWritten = AzureIoTJSONWriter_GetBytesUsed( &xWriter );
+
+    if( lBytesWritten < 0 )
+    {
+        LogDebug( ( "Error getting the bytes written for the properties confirmation JSON" ) );
+    }
+    else
+    {
+        LogDebug( ( "Sending acknowledged writable property. Payload: %.*s", lBytesWritten, ucPropertyPayloadBuffer ) );
+        xResult = AzureIoTHubClient_SendPropertiesReported( &xAzureIoTHubClient, ucPropertyPayloadBuffer, lBytesWritten, NULL );
+
+        if( xResult != eAzureIoTSuccess )
+        {
+            LogDebug( ( "There was an error sending the reported properties: 0x%08x", xResult ) );
+        }
+    }
+}
+
+/**
+ * @brief Property mesage callback handler
+ */
+static void prvHandleProperties( AzureIoTHubClientPropertiesResponse_t * pxMessage,
+                                 void * pvContext )
+{
+    ( void ) pvContext;
+
+    AzureIoTResult_t xResult;
+    double xIncomingTemperature;
+    uint32_t ulVersion;
+    bool xWasMaxTemperatureChanged = false;
+
+    switch( pxMessage->xMessageType )
+    {
+        case eAzureIoTHubPropertiesGetMessage:
+            LogInfo( ( "Device property document GET received" ) );
+            xResult = prvProcessProperties( pxMessage, eAzureIoTHubClientPropertyWritable, &xIncomingTemperature, &ulVersion );
+
+            if( xResult == eAzureIoTSuccess )
+            {
+                prvUpdateLocalProperties( xIncomingTemperature, ulVersion, &xWasMaxTemperatureChanged );
+                prvAckIncomingTemperature( xIncomingTemperature, ulVersion );
+            }
+            else
+            {
+                LogDebug( ( "There was an error processing incoming properties" ) );
+            }
+
+            if( xWasMaxTemperatureChanged )
+            {
+                prvSendNewMaxTemp( xIncomingTemperature );
+            }
+
+            break;
+
+        case eAzureIoTHubPropertiesWritablePropertyMessage:
+            LogInfo( ( "Device writeable property received" ) );
+            xResult = prvProcessProperties( pxMessage, eAzureIoTHubClientPropertyWritable, &xIncomingTemperature, &ulVersion );
+
+            if( xResult == eAzureIoTSuccess )
+            {
+                prvUpdateLocalProperties( xIncomingTemperature, ulVersion, &xWasMaxTemperatureChanged );
+                prvAckIncomingTemperature( xIncomingTemperature, ulVersion );
+            }
+            else
+            {
+                LogDebug( ( "There was an error processing incoming properties" ) );
+            }
+
+            if( xWasMaxTemperatureChanged )
+            {
+                prvSendNewMaxTemp( xIncomingTemperature );
+            }
+
+            break;
+
+        case eAzureIoTHubPropertiesReportedResponseMessage:
+            LogInfo( ( "Device reported property response received" ) );
+            break;
+
+        default:
+            LogError( ( "Unknown property message" ) );
+    }
```

Comment:
this is something configAssert(false)

Reply:
I don't think it's something we need to crash for.